base/applications/osk/settings.c

```diff
@@ -176,119 +76,107 @@ BOOL SaveDataToRegistry(VOID)
 
-    if (lResult != ERROR_SUCCESS)
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
-    }
+    /* Initialize the registry application settings */
+    Globals.bShowWarning = TRUE;
+    Globals.bIsEnhancedKeyboard = TRUE;
+    Globals.bAlwaysOnTop = TRUE;
+    Globals.bSoundClick = FALSE;
 
-    /* The value will be appended to the sound on click event */
-    dwValue = Globals.bSoundClick;
+    /* Set the coordinate values to default */
+    Globals.PosX = CW_USEDEFAULT;
+    Globals.PosY = CW_USEDEFAULT;
 
-    /* "Sound on Click" switcher value key */
-    lResult = RegSetValueExW(hKey,
-                             L"ClickSound",
-                             0,
-                             REG_DWORD,
-                             (BYTE *)&dwValue,
-                             sizeof(dwValue));
+    /* Warning dialog registry setting */
+    lResult = LoadDataFromRegistry(L"ShowWarning", &dwValue);
+    if (lResult != ERROR_FILE_NOT_FOUND)
+        Globals.bShowWarning = (dwValue != 0);
+
+    /* Enhanced keyboard switch dialog registry setting */
+    lResult = LoadDataFromRegistry(L"IsEnhancedKeyboard", &dwValue);
+    if (lResult != ERROR_FILE_NOT_FOUND)
+        Globals.bIsEnhancedKeyboard = (dwValue != 0);
+
+    /* Sound on click event registry setting */
+    lResult = LoadDataFromRegistry(L"ClickSound", &dwValue);
+    if (lResult != ERROR_FILE_NOT_FOUND)
+        Globals.bSoundClick = (dwValue != 0);
+
+    /* X coordinate dialog placement registry setting */
+    lResult = LoadDataFromRegistry(L"WindowLeft", &dwValue);
+    if (lResult != ERROR_FILE_NOT_FOUND)
+        Globals.PosX = dwValue;
+
+    /* Y coordinate dialog placement registry setting */
+    lResult = LoadDataFromRegistry(L"WindowTop", &dwValue);
+    if (lResult != ERROR_FILE_NOT_FOUND)
+        Globals.PosY = dwValue;
+
+    /* Top window state registry setting */
+    lResult = LoadDataFromRegistry(L"AlwaysOnTop", &dwValue);
+    if (lResult != ERROR_FILE_NOT_FOUND)
+        Globals.bAlwaysOnTop = (dwValue != 0);
+}
 
-    if (lResult != ERROR_SUCCESS)
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
-    }
+VOID SaveSettings(VOID)
+{
+    WINDOWPLACEMENT wp;
+    DWORD dwValue = 0;
 
-    /* The value will be appended to the X coordination dialog's placement */
-    dwValue = wp.rcNormalPosition.left;
+    /* Initialize the window placement structure */
+    wp.length = sizeof(WINDOWPLACEMENT);
+    GetWindowPlacement(Globals.hMainWnd, &wp);
 
-    /* Position X coordination of dialog's placement value key */
-    lResult = RegSetValueExW(hKey,
-                             L"WindowLeft",
-                             0,
-                             REG_DWORD,
-                             (BYTE *)&dwValue,
-                             sizeof(dwValue));
+    /* Warning dialog registry setting */
+    dwValue = Globals.bShowWarning;
```

Comment:
These assignments aren't necessary, you can just pass the value to the function directly.